src/index.js

```diff
@@ -287,17 +287,47 @@ export default function (babel) {
     return (typeof node.skinny === "boolean");
   }
 
+  // c/p babel-traverse/src/path/family.js getCompletionRecords
+  function getTailExpressions(path, allowLoops) {
+    let paths = [];
+
+    const add = function add(_path) {
+      if (_path) paths = paths.concat(getTailExpressions(_path));
+    };
+
+    if (path.isIfStatement()) {
+      add(path.get("consequent"));
+      add(path.get("alternate"));
+    } else if (path.isDoExpression()) {
+      add(path.get("body"));
+    } else if (allowLoops && (path.isFor() || path.isWhile())) {
```

Comment:
cc @rattrayalex 

I am not sure this implementation of the `allowLoops` flag is correct. If you pass `true`, you will get all the tail expressions in the loop (which is probably what you want) however if you pass `false` you will be treating the whole loop as a tail expression (which is almost definitely not what you want...)

Another check will probably need to be added below so that you don't push the loop to the return array.